src/librustc/hir/lowering.rs

```diff
@@ -865,7 +865,7 @@ impl<'a> LoweringContext<'a> {
             return;
         }
 
-        if self.in_scope_lifetimes.contains(&ident.modern()) {
+        if self.in_scope_lifetimes.contains(&ParamName::Plain(ident.modern())) {
```

Comment:
I don't think this check is correct anymore-- it's failing when it should succeed, causing lifetimes to be introduced multiple times (see the travis error). It should instead check if there is an in scope lifetime that uses the same identifier, right?

Reply:
Hmm. Is it the `ident.modern()` that's causing the problem, you think?

Reply:
I think I have a fix now